openedx/core/lib/gating/services.py

```diff
@@ -0,0 +1,47 @@
+"""
+A wrapper class around requested methods exposed in api.py
+"""
+import types
+
+from . import api as gating_api
+
+
+class GatingService(object):  # pylint: disable=too-few-public-methods
+    """
+    An xBlock service for xBlocks to talk to the Gating api.
+
+    NOTE: This is a Singleton class. We should only have one instance of it!
+    """
+
+    _instance = None
+
+    REQUESTED_FUNCTIONS = [
+        'get_gating_milestone_meta_info',
+        'get_gating_milestone',
+        'is_prereq_met'
+    ]
```

Comment:
Nit, and genuine question: Methods like `get_gating_milestone` leak the abstractions under them. Would the caller's lives be simpler if we had a different set of methods that hid that from them and presented them with a simpler interface?